RadixWallet/Clients/OnLedgerEntitiesClient/OnLedgerEntitiesClient+Interface.swift

```diff
@@ -285,3 +307,92 @@ extension OnLedgerEntitiesClient {
 		return dappMetadata
 	}
 }
+
+// MARK: - OnLedgerSyncOfAccounts
+public struct OnLedgerSyncOfAccounts: Sendable, Hashable {
+	/// Inactive virtual accounts, unknown to the Ledger OnNetwork.
+	public let inactive: IdentifiedArrayOf<Profile.Network.Account>
+	/// Accounts known to the Ledger OnNetwork, with state updated according to that OnNetwork.
+	public let active: IdentifiedArrayOf<Profile.Network.Account>
+}
+
+extension OnLedgerEntitiesClient {
+	/// returns the updated account, else `nil` if account was not changed,
+	public func syncThirdPartyDepositWithOnLedgerSettings(
+		account: Profile.Network.Account
+	) async throws -> Profile.Network.Account? {
+		guard let ruleOfAccount = try await getOnLedgerCustomizedThirdPartyDepositRule(addresses: [account.address]).first else {
+			return nil
+		}
+		let current = account.onLedgerSettings.thirdPartyDeposits.depositRule
+		if ruleOfAccount.rule == current {
+			return nil
+
+		} else {
+			var account = account
+			account.onLedgerSettings.thirdPartyDeposits.depositRule = ruleOfAccount.rule
+			return account
+		}
+	}
+
+	public func syncThirdPartyDepositWithOnLedgerSettings(
+		addressesOf accounts: IdentifiedArrayOf<Profile.Network.Account>
+	) async throws -> OnLedgerSyncOfAccounts {
+		let activeAddresses: [CustomizedOnLedgerThirdPartDepositForAccount]
+		do {
+			activeAddresses = try await getOnLedgerCustomizedThirdPartyDepositRule(addresses: accounts.map(\.accountAddress))
+		} catch is GatewayAPIClient.EmptyEntityDetailsResponse {
+			return OnLedgerSyncOfAccounts(inactive: accounts, active: [])
+		} catch {
+			throw error
+		}
+		var inactive: IdentifiedArrayOf<Profile.Network.Account> = []
+		var active: IdentifiedArrayOf<Profile.Network.Account> = []
+		for account in accounts { // iterate with `accounts` to retain insertion order.
+			if let onLedgerActiveAccount = activeAddresses.first(where: { $0.address == account.address }) {
+				var activeAccount = account
+				activeAccount.onLedgerSettings.thirdPartyDeposits.depositRule = onLedgerActiveAccount.rule
+				active.append(activeAccount)
+			} else {
+				inactive.append(account)
+			}
+		}
+		return OnLedgerSyncOfAccounts(inactive: inactive, active: active)
+	}
+
+	public struct CustomizedOnLedgerThirdPartDepositForAccount: Sendable, Hashable {
+		public let address: AccountAddress
+		public let rule: Profile.Network.Account.OnLedgerSettings.ThirdPartyDeposits.DepositRule
+	}
+
+	public func getOnLedgerCustomizedThirdPartyDepositRule(
+		addresses: some Collection<AccountAddress>
+	) async throws -> [CustomizedOnLedgerThirdPartDepositForAccount] {
+		try await self.getAccounts(
+			Array(addresses),
+			// actually we wanna `resourceMetadataKeys` as well here, but we cannot since
+			// the count will exceed `EntityMetadataKey.maxAllowedKeys`.
```

Comment:
Do we need all of the `resourceMetadataKeys`? Else we could read up to 8 of those keys, and these two.

Reply:
I guess we want them yes, I will remove this old comment, it is not really an issue now since I do not write the downloaded metadata to the cache.